### PR TITLE
6230751: [Fmt-Ch] Recursive MessageFormats in ChoiceFormats ignore indicated subformats

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR which updates an incorrect code example in _java/text/ChoiceFormat_.

ChoiceFormat (and MessageFormat) provide an example of how to produce a pattern that supports singular and plural forms. The ChoiceFormat example is incorrect, as recursive MessageFormats produced by a ChoiceFormat subformat only recognize subformats defined through the MessageFormat pattern syntax, not through the subformats contained within the top level MessageFormat. 

In the original example,
`Format[] testFormats = {fileform, null, NumberFormat.getInstance()};` could have been replaced with 
`Format[] testFormats = {fileform, null, new ChoiceFormat("0#BROKEN")};` and the original output would have been the same.

This PR replaces the example with the one used in MessageFormat, which is correct. 

This PR also includes a drive-by fix to remove leftover `<blockquote>`s from a previous `@snippet` conversion.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8321420](https://bugs.openjdk.org/browse/JDK-8321420) to be approved

### Issues
 * [JDK-6230751](https://bugs.openjdk.org/browse/JDK-6230751): [Fmt-Ch] Recursive MessageFormats in ChoiceFormats ignore indicated subformats (**Bug** - P4)
 * [JDK-8321420](https://bugs.openjdk.org/browse/JDK-8321420): [Fmt-Ch] Recursive MessageFormats in ChoiceFormats ignore indicated subformats (**CSR**)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16891/head:pull/16891` \
`$ git checkout pull/16891`

Update a local copy of the PR: \
`$ git checkout pull/16891` \
`$ git pull https://git.openjdk.org/jdk.git pull/16891/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16891`

View PR using the GUI difftool: \
`$ git pr show -t 16891`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16891.diff">https://git.openjdk.org/jdk/pull/16891.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16891#issuecomment-1832821303)

### CODE REVIEW

**Review by naotoj — COMMENTED** [review]
Do you think it would be helpful if we describe that limitation so that such confusion will not arise?

**Comment by justin-curtis-lu** [comment]
> Do you think it would be helpful if we describe that limitation so that such confusion will not arise?

Yes, that's a good point. I didn't initially include one because I wanted to look into [JDK-4270867](https://bugs.openjdk.org/browse/JDK-4270867) (which would be the fix to the limitation) first.

File: src/java.base/share/classes/java/text/MessageFormat.java [hunk 1]
```diff
@@ -297,24 +287,27 @@
...
- * </blockquote>
  *
  * <p>
- * <strong>Note:</strong> As we see above, the string produced
- * by a {@code ChoiceFormat} in {@code MessageFormat} is treated as special;
- * occurrences of '{' are used to indicate subformats, and cause recursion.
+ * <strong id="pattern_caveats">Notes:</strong> As seen in the previous snippet,
+ * the string produced by a {@code ChoiceFormat} in {@code MessageFormat} is
+ * treated as special; occurrences of '{' are used to indicate subformats, and
+ * cause recursion. A limitation to this behavior is that, if a {@code MessageFormat}
```
Comment (justin-curtis-lu) [inline src/java.base/share/classes/java/text/MessageFormat.java:299]: @naotoj Please let me know if the added limitations wording reads clearly to you. I would rather not have a whole dedicated code example to explain a _limitation_, but if needed for better understanding, I can add one.

Comment (naotoj) [reply src/java.base/share/classes/java/text/MessageFormat.java:299]: Do you mean `{@code MessageFormat}{@code FormatElement}`?
Also, I would rather remove "A limitation to this behavior is that," as readers would know the following explanation explains a limitation.

Comment (justin-curtis-lu) [reply src/java.base/share/classes/java/text/MessageFormat.java]: Right, no need for `{@code MessageFormat}` as it's already implied. Also removed "_A limitation to this behavior is that_" as suggested.

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**